Add User Creation and Authentication

alexdanieldm opened the pull request:
Using firebase methods enable app to: 

- On `logIn` screen - authenticate users with `createUserWithEmailAndPassword()`
- On `signUp` screen - create user with `signInWithEmailAndPassword()`
- On `signUp` screen - log out user with `signOut()`

All while preserving User Credentials for all the App with `onAuthStateChanged()`

alexdanieldm committed:
feat(config.js): connect to firebase

chore: add firebase folder

chore: to signUp a logIn inline link

alexdanieldm committed:
refactor:  add loading component

fix: install firebase properly

build: run ./gradlew clean on android folder

alexdanieldm committed:
refactor(todo.js): to promises

chore(todo.js): reorder functions

alexdanieldm committed:
fix(logIn.js): fullName TextInput receive strings

feat(signUp.js): add user creation with createUserWithEmailAndPassword

alexdanieldm committed:
build(package.json): new script to clean gradle

alexdanieldm committed:
feat(signUp.js): redirect user to logIn screen after successful sign up

alexdanieldm committed:
feat(todo.js): persist ToDo items after reloads using AsyncStorage

alexdanieldm committed:
feat(logIn.js): add user athentication

fix(signUp.js): now only redirect on successful sign up

perf(signUp.js): to use error.message on alerts

duranmla commented on src/firebase/config.js:
```diff
@@ -0,0 +1,20 @@
+import '@firebase/auth';
+import '@firebase/firestore';
+
+const firebaseConfig = {
```
All this configuration is okay to be here? Often the credentials for things that are not public matter needs to use environment variables. Are you sure the documentation doesn't suggest something around environment variables or does it suggest committing this to git?

alexdanieldm replied:
None of the documentation that i check say something about committing or not the values to a remote repo, nor does it mention environment variables; that being said, it make sense to use them for this project. I'll make sure to add a `.env`

alexdanieldm committed:
fix(package-lock): to lockfileVersion@1

feat(App): implement onAuthStateChanged() to preserve user credentials

chore(logIn): to more user friendly alert message message

alexdanieldm committed:
refactor!: to environment variables

BREAKING CHANGE: code on public repo now only work if a new .env file its properly set up using the env.template

alexdanieldm committed:
feat(LogWithTime): create function to output console log message and states values with time stamps
* This is for a better developer expirence and do not affect regular users

chore: add utils folder

alexdanieldm committed:
feat(App): now show Loading component while checking if user is already SignIn

refactor(App): to protected Todo route from non-authenticated users

alexdanieldm committed:
chore: dismiss Keyboard after pressing button on logIn and signUp screens

chore: to more descriptive code

alexdanieldm committed:
feat(App): implement firebase  signOut() method

feat: create and implement logOutButton

chore(logIn): remove unnecessary navigate method
